aip/general/0203.md

```diff
@@ -44,42 +44,51 @@ document this behavior for clients.
 
 ## Vocabulary
 
-### Required
+### Identifier
 
-The use of `REQUIRED` indicates that the field **must** be present (and set to
-a non-empty value) on the request or resource.
+The use of `IDENTIFIER` indicates that a field within a resource message is used
+to identify the resource. It **must** be attached to the `name` field and any
+other fields within a resource that are used in identifying the resource within
+a collection. See [fields representing resource names].
 
-A field **should** only be described as required if _either_:
+This annotation **should not** be applied to references to other resources within a
+message.
 
-- It is a field on a resource that a user provides somewhere as input. In this
-  case, the resource is only valid if a "truthy" value is _stored_.
-  - When [creating][aip-133] the resource, a value **must** be provided for the
-    field on the create request.
-  - When [updating][aip-134] the resource, the user **may** omit the field
-    provided that the field is also absent from the field mask, indicating no
-    change to the field (otherwise it **must** be provided).
-- It is a field on a request message (a message that is an argument to an RPC,
-  with a name usually ending in `Request`). In this case, a value **must** be
-  provided as part of the request, and failure to do so **must** cause an error
-  (usually `INVALID_ARGUMENT`).
+### Immutable
 
-We define the term "truthy" above as follows:
+The use of `IMMUTABLE` indicates that a field on a resource cannot be changed
+after it's creation. This can apply to either fields that are input or outputs,
+required or optional.
 
-- For primitives, values other than `0`, `0.0`, empty string/bytes, and `false`
-- For repeated fields maps, values with at least one entry
-- For messages, any message with at least one "truthy" field.
+When a service receives an immutable field in an update request (or similar),
+even if included in the update mask, the service **should** ignore the field if
+the value matches, but **should** error with `INVALID_ARGUMENT` if a change is
+requested.
 
-Fields **should not** be described as required in order to signify:
+Potential use cases for immutable fields (this is not an exhaustive list) are:
 
-- A field which will always be present in a response.
-- A field which is conditionally required in some situations.
-- A field on any message (including messages that are resources) which is never
-  used as user input.
+- Names or IDs which are set on creation and then used as a primary key.
```

Comment:
perhaps I'm reading this wrong, but are we saying that IDENTIFIER fields could also be marked as IMMUTABLE? names shouldn't be mutable but the IDENTIFIER field should be enough to infer that.

Reply:
ack, yeah this should be updated. removed the identifier use case, replaced it with something more practical.

Yes, agreed: you should only need `IDENTIFIER`.